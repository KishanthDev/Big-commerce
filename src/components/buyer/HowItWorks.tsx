export default function HowItWorks() {
    const steps = [
      {
        number: '1',
        title: 'Sign Up',
        description: 'Create your free account in minutes to start browsing our curated marketplace.'
      },
      {
        number: '2',
        title: 'Browse Products',
        description: 'Explore categories and discover sellers offering unique products in your area.'
      },
      {
        number: '3',
        title: 'Start Shopping',
        description: 'Purchase directly from sellers with secure checkout and local delivery options.'
      }
    ];
  
    return (
      <section className="bg-gray-50 dark:bg-black py-16">
        <div className="container mx-auto px-5">
          <h2 className="text-3xl font-semibold dark:text-white text-center mb-12">How It Works: 3 Simple Steps</h2>
          <div className="max-w-4xl mx-auto grid grid-cols-1 md:grid-cols-3 gap-8">
            {steps.map((step, index) => (
              <div key={index} className="text-center">
                <div className="bg-blue-600 text-white w-10 h-10 rounded-full flex items-center justify-center mx-auto mb-5 font-bold">
                  {step.number}
                </div>
                <h3 className="text-xl dark:text-white font-semibold mb-3">{step.title}</h3>
                <p className="text-gray-600 dark:text-white text-sm">{step.description}</p>
              </div>
            ))}
          </div>
        </div>
      </section>
    );
  }