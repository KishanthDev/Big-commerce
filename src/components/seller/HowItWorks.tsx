export default function HowItWorks() {
    const steps = [
      {
        number: '1',
        title: 'Sign Up',
        description: 'Create your free account in minutes.Allyou need is basic information to get started.'
      },
      {
        number: '2',
        title: 'Upload Products',
        description: 'Add your inventory with photos ,description and pricing.User-friendly dashboard makes this way.'
      },
      {
        number: '3',
        title: 'Start Selling',
        description: 'Begin recieving orders from local customers.Mange everything from your personalised dashboard'
      }
    ];
  
    return (
      <section className="bg-gray-50 dark:bg-black py-16">
        <div className="container mx-auto px-5">
          <h2 className="text-3xl font-semibold dark:text-white text-center mb-12">How It Works: 3 Simple Steps</h2>
          <div className="max-w-4xl mx-auto grid grid-cols-1 md:grid-cols-3 gap-8">
            {steps.map((step, index) => (
              <div key={index} className="text-center">
                <div className="bg-blue-600 text-white w-10 h-10 rounded-full flex items-center justify-center mx-auto mb-5 font-bold">
                  {step.number}
                </div>
                <h3 className="text-xl dark:text-white font-semibold mb-3">{step.title}</h3>
                <p className="text-gray-600 dark:text-white text-sm">{step.description}</p>
              </div>
            ))}
          </div>
        </div>
      </section>
    );
  }