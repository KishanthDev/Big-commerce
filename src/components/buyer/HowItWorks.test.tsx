import { render, screen } from "@testing-library/react";
import HowItWorks from "./HowItWorks";

describe("HowItWorks Component", () => {
  const mockSteps = [
    {
      number: "1",
      title: "Sign Up",
      description:
        "Create your free account in minutes to start browsing our curated marketplace.",
    },
    {
      number: "2",
      title: "Browse Products",
      description:
        "Explore categories and discover sellers offering unique products in your area.",
    },
    {
      number: "3",
      title: "Start Shopping",
      description:
        "Purchase directly from sellers with secure checkout and local delivery options.",
    },
  ];

  test("displays the main heading", () => {
    render(<HowItWorks />);
    expect(
      screen.getByRole("heading", { name: /How It Works: 3 Simple Steps/i }),
    ).toBeInTheDocument();
  });

  test("displays correct step information", () => {
    render(<HowItWorks />);

    mockSteps.forEach((step) => {
      expect(screen.getByText(step.number)).toBeInTheDocument();
      expect(screen.getByText(step.title)).toBeInTheDocument();
      expect(screen.getByText(step.description)).toBeInTheDocument();
    });
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<HowItWorks />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("has proper accessibility attributes", () => {
    render(<HowItWorks />);
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
      "How It Works: 3 Simple Steps",
    );
    expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(
      mockSteps.length,
    );
  });
});
