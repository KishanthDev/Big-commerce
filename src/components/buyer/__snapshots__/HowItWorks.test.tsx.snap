// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowItWorks Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="bg-gray-50 dark:bg-black py-16"
  >
    <div
      class="container mx-auto px-5"
    >
      <h2
        class="text-3xl font-semibold dark:text-white text-center mb-12"
      >
        How It Works: 3 Simple Steps
      </h2>
      <div
        class="max-w-4xl mx-auto grid grid-cols-1 md:grid-cols-3 gap-8"
      >
        <div
          class="text-center"
        >
          <div
            class="bg-blue-600 text-white w-10 h-10 rounded-full flex items-center justify-center mx-auto mb-5 font-bold"
          >
            1
          </div>
          <h3
            class="text-xl dark:text-white font-semibold mb-3"
          >
            Sign Up
          </h3>
          <p
            class="text-gray-600 dark:text-white text-sm"
          >
            Create your free account in minutes to start browsing our curated marketplace.
          </p>
        </div>
        <div
          class="text-center"
        >
          <div
            class="bg-blue-600 text-white w-10 h-10 rounded-full flex items-center justify-center mx-auto mb-5 font-bold"
          >
            2
          </div>
          <h3
            class="text-xl dark:text-white font-semibold mb-3"
          >
            Browse Products
          </h3>
          <p
            class="text-gray-600 dark:text-white text-sm"
          >
            Explore categories and discover sellers offering unique products in your area.
          </p>
        </div>
        <div
          class="text-center"
        >
          <div
            class="bg-blue-600 text-white w-10 h-10 rounded-full flex items-center justify-center mx-auto mb-5 font-bold"
          >
            3
          </div>
          <h3
            class="text-xl dark:text-white font-semibold mb-3"
          >
            Start Shopping
          </h3>
          <p
            class="text-gray-600 dark:text-white text-sm"
          >
            Purchase directly from sellers with secure checkout and local delivery options.
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
